server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -371,20 +377,26 @@ public CompletionStage<Result> previousWithApplicantId(
 
               if (block.isPresent()) {
                 ApplicantPersonalInfo personalInfo = applicantStage.toCompletableFuture().join();
-                return ok(
-                    editView.render(
-                        buildApplicationBaseViewParams(
-                            request,
-                            applicantId,
-                            programId,
-                            blockId,
-                            inReview,
-                            roApplicantProgramService,
-                            block.get(),
-                            personalInfo,
-                            ApplicantQuestionRendererParams.ErrorDisplayMode.HIDE_ERRORS,
-                            applicantRoutes,
-                            profile)));
+                ApplicationBaseView.Params applicationParams =
```

Comment:
This is something I assume might change down the road (or in this PR 🤷 ). We need basically the same info for the two views. It's probably worth moving ApplicationBaseView.Params into a utility class somewhere, since ApplicationBaseView is a superclass of ApplicantProgramBlockEditView and I haven't made the NorthStar view extend those classes since we don't need the utility rendering functions, so it's a little odd to reference from the north star view. Open to opinions.

Reply:
At first glance I like this pattern of using a data object to pass the necessary info the the view, it keeps the method signatures a little cleaner.

otoh, it does make it a little harder to tell what's actually being used, and I don't think IntelliJ is smart enough to notice that a field on the data object is unused.

if we keep these data objects, I'd agree that moving them into a class on their own makes sense.

Reply:
This PR seems big enough already to me, so I'll plan to factor it out in a follow up, since it's referenced in a bunch of places